messybrainz/db/common.py

```diff
@@ -0,0 +1,108 @@
+
+    return clusters_modified, clusters_added_to_redirect
+
+
+def create_entity_clusters_for_anomalies(connection,
+                                        fetch_entities_left_to_cluster,
+                                        get_entity_gids_from_recording_json_using_mbids,
+                                        get_cluster_id_using_msid,
+                                        link_entity_mbid_to_entity_cluster_id):
+    """Creates entity clusters for the anomalies (A single MSID pointing
+       to multiple MBIDs in entity_redirect table).
+
+    Args:
+        connection: the sqlalchemy db connection to be used to execute queries
+        fetch_entities_left_to_cluster(function): Returns mbids for the entity MBIDs that
+                                                were not clustered after executing the
+                                                first phase of clustering (clustering without
+                                                considering anomalies). These are anomalies
+                                                (A single MSID pointing to multiple MBIDs in
+                                                entity_redirect table).
+        get_entity_gids_from_recording_json_using_mbids(function): Returns entity MSIDs using
+                                                                an entity MBID.
+        get_cluster_id_using_msid(function): Gets the cluster ID for a given MSID.
+        link_entity_mbid_to_entity_cluster_id(function): Links the entity mbid to the cluster_id.
+
+    Returns:
+        clusters_add_to_redirect (int): number of clusters added to redirect table.
+    """
+
+    clusters_add_to_redirect = 0
+    entities_left = fetch_entities_left_to_cluster(connection)
+    for entity_mbid in entities_left:
+        entity_gids = get_entity_gids_from_recording_json_using_mbids(connection, entity_mbid)
+        cluster_ids = {get_cluster_id_using_msid(connection, entity_gid) for entity_gid in entity_gids}
+        for cluster_id in cluster_ids:
+            link_entity_mbid_to_entity_cluster_id(connection, cluster_id, entity_mbid)
+            clusters_add_to_redirect += 1
+
+    return clusters_add_to_redirect
+
+
+def create_entity_clusters_without_considering_anomalies(connection,
+                                                        fetch_unclustered_entity_mbids,
+                                                        fetch_unclustered_gids_for_entity_mbids,
+                                                        get_entity_cluster_id_using_entity_mbids,
+                                                        link_entity_mbids_to_entity_cluster_id,
+                                                        insert_entity_cluster):
+    """Creates cluster for entity without considering anomalies (A single MSID pointing
+       to multiple MBIDs in entity_redirect table).
+
+    Args:
+        connection: the sqlalchemy db connection to be used to execute queries.
```

Comment:
taking a connection as input here and then creating a connection in the function as well.

Reply:
done:). removed creating a connection in the function.